dropwizard-lifecycle/src/main/java/io/dropwizard/lifecycle/setup/LifecycleEnvironment.java

```diff
@@ -61,6 +67,25 @@ public ExecutorServiceBuilder executorService(String nameFormat, ThreadFactory f
         return new ExecutorServiceBuilder(this, nameFormat, factory);
     }
 
+    public ExecutorService virtualExecutorService(String nameFormat) {
+        try {
+            Object virtualThreadBuilder = Thread.class.getDeclaredMethod("ofVirtual").invoke(null);
+            Object virtualThreadFactory = Class.forName("java.lang.Thread$Builder")
+                .getDeclaredMethod("factory")
+                .invoke(virtualThreadBuilder);
+            InstrumentedThreadFactory factory = new InstrumentedThreadFactory((ThreadFactory) virtualThreadFactory, getMetricRegistry(), nameFormat);
+            ExecutorService virtualThreadExecutor = (ExecutorService) Executors.class
+                .getDeclaredMethod("newThreadPerTaskExecutor", ThreadFactory.class)
+                .invoke(null, factory);
+            manage(new ExecutorServiceManager(virtualThreadExecutor, Duration.seconds(5), nameFormat));
```

Comment:
I'd suggest to return an `ExecutorServiceBuilder`. This will allow customizations of the `ExecutorService` and automatically manage it in the `build()` method.

Reply:
> I didn't create/reuse any builders as all the fields in the ExecutorServiceBuilders are obsolete for virtual threads. We let JVM handle it all instead.

If the concern is consistency, it will be rather complicated. You can't really set min/max threads with virtual threads for example. Or maybe you can by manipulating the ThreadFactory but there wouldn't be much point to it really conceptually. Java Executors itself does not really provide that kind of option. It's only `newThreadPerTaskExecutor` which is basically "let virtual thread pool of JVM handle everything".

Reply:
I'm relatively new to virtual threads and just wanted to ask, if we can modify the underlying executor of platform threads. If there is no such option, we indeed can't use the `ExecutorServiceBuilder` for this purpose.

I just wanted to avoid the 'magic' number with 5 seconds as the shutdown time. I'd suggest to provide an overload to allow customizations of that parameter then.

Reply:
yeah, from what I can see there isn't much we can change when it comes to virtual thread pool management.

I will add an overload for the shutdown time.